### PR TITLE
[Enhancement]error message for unsupported file format

### PR DESCRIPTION
**Pull request by zombee0**
## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes:

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Review by Smith-Cruise — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetastoreApiConverter.java [hunk 1]
```diff
@@ -168,10 +168,14 @@ public static HudiTable toHudiTable(Table table, String catalogName) {
 
     public static Partition toPartition(StorageDescriptor sd, Map<String, String> params) {
         requireNonNull(sd, "StorageDescriptor is null");
+        RemoteFileInputFormat fileFormat = toRemoteFileInputFormat(sd.getInputFormat());
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetastoreApiConverter.java:171]: it's better to check this in the `toRemoteFileInputFormat`

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 3 / 4 (75.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastoreApiConverter.java|3|4|75.00%|[173]|

File: fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetastoreApiConverter.java [hunk 2]
```diff
@@ -168,10 +168,14 @@ public static HudiTable toHudiTable(Table table, String catalogName) {
...
+        if (fileFormat == null) {
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetastoreApiConverter.java:172]: It also work for unpartitioned table?

File: fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetastoreApiConverter.java (hunk 1 above)
Comment (zombee0) [reply fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetastoreApiConverter.java]: done

File: fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetastoreApiConverter.java (hunk 2 above)
Comment (zombee0) [reply fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetastoreApiConverter.java]: yes, both partitioned and unpartitioned table.

**Review by stephen-shelby — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 3 / 4 (75.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hive/RemoteFileInputFormat.java|3|4|75.00%|[45]|

**Review by imay — APPROVED** [review]

**Pull request merged**